### PR TITLE
[SPARK-46744][SPARK-SHELL][SQL][CONNECT][PYTHON][R] Display clear `exit command` for all spark terminal

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to display clear `exit command` for all spark terminal, includes: 
`spark-shell`, `spark-sql`, `spark-connect-repl`, `pyspark` and `sparkR`.

### Why are the changes needed?
Because some `exit commands` are `different` across different terminals, in order to promote a better user experience.

### Does this PR introduce _any_ user-facing change?
Yes, when the user starts the terminal, they can clearly see `the exit command`.

### How was this patch tested?
Manually test.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by panbingkun** [comment]
- spark-shell
before:


after:


- spark-sql
before:

after:

- spark-connect-repl
before:

after:
image

**Comment by panbingkun** [comment]
cc @HyukjinKwon

File: connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/application/ConnectRepl.scala [hunk 1]
```diff
@@ -45,7 +45,8 @@ object ConnectRepl {
       | ___/ / /_/ / /_/ / /  / ,<    / /___/ /_/ / / / / / / /  __/ /__/ /_
       |/____/ .___/\__,_/_/  /_/|_|   \____/\____/_/ /_/_/ /_/\___/\___/\__/
       |    /_/
-      |""".stripMargin
+      |
+      |Use exit or Ctrl-D (i.e. EOF) to exit.""".stripMargin
```
Comment (HyukjinKwon) [inline connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/application/ConnectRepl.scala:49]: You gotta fix `shell.py` and `shell.R` too.

Comment (panbingkun) [reply connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/application/ConnectRepl.scala:49]: Okay, let me do it.

Comment (panbingkun) [reply connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/application/ConnectRepl.scala:49]: Done.

**Pull request closed without merging**